File: js/task3c/S3_PIPE_02.js
function compose(...fns) {
  return function(initialValue) {
    return fns.reduceRight((acc, fn) => fn(acc), initialValue);
  };
}
const add1 = x => x + 1;
const double = x => x * 2;
const composed = compose(add1, double);
console.log(composed(5));

File: python/task3c/S3_PIPE_02.py
def compose(*fns):
  def inner(initial_value):
      result = initial_value
      for fn in reversed(fns):
          result = fn(result)
      return result
  return inner

add1 = lambda x: x + 1
double = lambda x: x * 2

composed = compose(add1, double)
print(composed(5))
